### ISSUE TITLE
Automated code testing using CI/CD

### ISSUE THREAD

**Issue by Alaz-Oz** [2G-Labs/DSA#32]
There are actual codes in the repo (example codes). Those codes should be checked by the compiler and unit tests, so that there aren't any errors when we push any commit to the main branch.

So, I propose we use CI/CD for auto testing these programs for any errors.

The tests should perform:
- **Compilation check**: Verifying if the program compiles successfully.
- **Test case checks**: Verifying if the program does what it is expected to do.

The `compilation check` don't require any changes in the repo. But the `Test case checks` needs:
- **Input**: Input for the running tests (input data)
- **Expected Output**: What should the program output (if the output of the program doesn't matches the `expected output`, then the test case fails.

Majority of the program will have the input hardcoded (written in their source code), so only the `Expected output` is of concern.

So, how should we provide the expected output in the repo.

I propose we include a output.txt file for each program

**Comment by 2G-Afroz** [comment]
Yep, we should.

**Comment by Alaz-Oz** [comment]
@2G-Afroz How would you like to add the `expected output` in the repo?
I could see 2 ways:
- Add standard comment in the cpp file showing output.
- Include a separate file that shoes output. (Every Code should have one then)

> ![Note]
> We may temporarily make this repo c++ only, and further add other lang support as we go.

**Comment by 2G-Afroz** [comment]
we can include separate .txt file which will contain output of the code.
We will use same example in different codes so that there will only be one .txt(output) file.

**Comment by Alaz-Oz** [comment]
What you meant by
> We will use same example in different codes so that there will only be one .txt(output) file.

Does that means that different example codes (written in different programming lang) of the same algorithm will have single output?

**Comment by 2G-Afroz** [comment]
> Does that means that different example codes (written in different programming lang) of the same algorithm will have single output?
Yes

**Comment by 2G-Afroz** [comment]
And we will mention that `example list/array` in `Standard.md` file.

**Comment by Alaz-Oz** [comment]
> And we will mention that example list/array in Standard.md file.

Elaborate this.

**Comment by 2G-Afroz** [comment]
I want to say that we will mention the `example list/array (like [ 2, 1, 7, -3, 9, 5, 19])` in `Standard.md` file.
So, whenever someone would want to contribute they will read `Standard.md` file and there they will get the example which they will have to use in `Code Example`.

**Comment by Alaz-Oz** [comment]
Nope, that is not right. Different algo have different needs, like sorted/unsorted. So it is not right to be included in STANDARD.

## My idea
- Input that program will need in `input.txt` in the specific algorithm dir.
- Output (Whatever the code will output) in `output.txt` in the same dir.
- If there will be multiple codes then name will be like `output1.txt`, `output2.txt`, ...

## Benefits
- **Easier to maintain**: Independent `io` (input and output) of the program.
- **Freedom for dev**: Coder could showcase a suitable test case, that demonstrate the code better.
- **CI/CD friendly**: Easier to implement in CI/CD.
